### ISSUE TITLE
libnacl, mingw and Windows

### ISSUE THREAD

**Issue by indraastra** [saltstack/libnacl#19]
TL;DR **init**.py looks for libsodium.dll, but the prebuilt dll is libsodium-10.dll. Easy fix is to rename it but I don't know what that might break. Alternate fix is to make dependency lookup version-independent.

Here is my setup story on both a Mac and Windows machine, for posterity.

I first trialed libnacl on a Mac dev box and when I tried to pip install libnacl, I got an error saying I was missing a nacl lib. In fact, this was my first time even hearing about libsodium (I was using tweetnacl) but thankfully, it was only a `brew install libsodium` away and I was up and running before long. Great! 

However...

For many unsavory reasons, I do most of my development in Windows, so I now had to figure out how to satisfy the nacl dependency without brew. I was using a weird mix of cygwin and msys so I wasn't entirely sure where to place a dll even if I got my hands on one.

I noticed libsodium had prebuilt releases for mingw, so I jettisoned my cygwin setup and started with mingw from scratch. After finding a dll named libsodium-10.dll in the libsodium release tarball, I figured I'd place it in C:\MinGW\bin and that would be that. Well, **init**.py looks specifically for libsodium.dll so the dll had to be renamed before it was happy.

It would be an easy fix to make the lookup in python a less version-dependent but I don't know how safe that is in both a security and dependency sense.
